services/src/main/java/org/fao/geonet/api/processing/MInspireEtfValidateProcess.java

```diff
@@ -0,0 +1,209 @@
+
+        try {
+            this.probeName = new ObjectName(String.format("geonetwork:name=batch-etf-inspire,idx=%s", this.hashCode()));
+        } catch (MalformedObjectNameException e) {
+            e.printStackTrace();
+        }
+    }
+
+    public void deleteAll() {
+        runInNewTransaction("minspireetfvalidate-deleteall", new TransactionTask<Object>() {
+            @Override
+            public Object doInTransaction(TransactionStatus transaction) throws Throwable {
+                deleteAllDate = System.currentTimeMillis();
+                //urlAnalyser.deleteAll();
+                return null;
+            }
+        });
+    }
+
+    public void processMetadata(Set<String> uuids) throws Exception {
+        IMetadataUtils metadataRepository = appContext.getBean(IMetadataUtils.class);
+        MetadataValidationRepository metadataValidationRepository = appContext.getBean(MetadataValidationRepository.class);
+        AccessManager accessManager = appContext.getBean(AccessManager.class);
+        InspireValidatorUtils inspireValidatorUtils = appContext.getBean(InspireValidatorUtils.class);
+        SchemaManager schemaManager = appContext.getBean(SchemaManager.class);
+        DataManager dataManager = appContext.getBean(DataManager.class);
+
+        metadataToAnalyseCount = uuids.size();
+        analyseMdDate = System.currentTimeMillis();
+
+        for (String uuid : uuids) {
+            if (!metadataRepository.existsMetadataUuid(uuid)) {
+                metadataAnalysed++;
+                metadataNotAllowed++;
+                continue;
+            }
+
+            for (AbstractMetadata record : metadataRepository.findAllByUuid(uuid)) {
+                try {
+                    if (!accessManager.canEdit(serviceContext, String.valueOf(record.getId()))) {
+                        metadataNotAllowed++;
+                    } else {
+                        runInNewTransaction("minspireetfvalidate-process-metadata", new TransactionTask<Object>() {
+                            @Override
+                            public Object doInTransaction(TransactionStatus transaction) throws Throwable {
+                                // Evaluate test conditions for INSPIRE test suites, to apply to the metadata
+                                Map<String, String> testsuiteConditions = inspireValidatorUtils.getTestsuitesConditions();
+
+                                try {
+                                    boolean inspireMetadata = false;
+
+                                    for (Map.Entry<String, String> entry : testsuiteConditions.entrySet()) {
+                                        boolean applyCondition = Xml.selectBoolean(record.getXmlData(false),
+                                            entry.getValue(),
+                                            schemaManager.getSchema(record.getDataInfo().getSchemaId()).getNamespaces());
+                                        if (applyCondition) {
```

Comment:
Maybe we should index that. This means this record is out of the scope of INSPIRE probably.
So we could add a new validation status MetadataValidationStatus.DOES_NOT_APPLY and in this case do not display the status icon in search results ? So we would end up with 4 states : green/red/grey/blank. What do you think ?